contract/services.py

```diff
@@ -75,15 +87,71 @@ def __evaluate_contract_valuation(self, contract_details_result):
         )
         return result_contract_valuation["data"]["total_amount"]
 
+    # TODO update contract scenario according to wiki page
+    @check_authentication
+    def update(self, contract):
+        try:
+            # check rights for contract / amendments
+            if not (self.user.has_perms(ContractConfig.gql_mutation_update_contract_perms) or self.user.has_perms(
+                    ContractConfig.gql_mutation_approve_ask_for_change_contract_perms)):
+                raise PermissionError("Unauthorized")
+            updated_contract = ContractModel.objects.filter(id=contract['id']).first()
+            # updatable scenario
+            if self.__check_rights_by_status(updated_contract.state) == "updatable":
+                if "code" in contract:
+                    raise ContractUpdateError("That fields are not editable in that permission")
+                return _output_result_success(
+                    dict_representation=self.__update_contract_fields(
+                        contract_input=contract,
+                        updated_contract=updated_contract
+                    )
+                )
+            # approvable scenario
```

Comment:
check satus: if in “Negotiable” changes are possible only with the authority “Approve/ask for change”